README.md
<h1 align="center">
    <img src=".github/modelo-boletim-escolar.png" width="400" alt="Logo BoletimEscolar">
</h1>
<p align="center">
    <img alt="Plataforma" src="https://img.shields.io/static/v1?label=Plataforma&message=Web&color=000000&labelColor=00E88F">
    <img alt="BoletimEscolar" src="https://img.shields.io/static/v1?label=Version&message=3.26.69&color=000000&labelColor=00E88F">
    <img alt="Tamanho do repositório" src="https://img.shields.io/github/repo-size/NyctibiusVII/BoletimEscolar?color=000000&labelColor=00E88F">
    <a href="https://github.com/NyctibiusVII/BoletimEscolar/blob/main/LICENSE">
        <img alt="Licença" src="https://img.shields.io/static/v1?label=License&message=MIT&color=000000&labelColor=00E88F">
    </a>
</p>
<p align="center">
    <a href="#boletimescolar-">Projeto</a>&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;
    <a href="#tecnologias-">Tecnologias</a>&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;
    <a href="#layout-">Layout</a>&nbsp;&nbsp;&nbsp;|&nbsp;&nbsp;&nbsp;
    <a href="#licença-%EF%B8%8F">Licença</a>
</p>

# BoletimEscolar <img src=".github/favicon.svg" width="32" alt="favicon">
Geração de Boletim escolar automatizado e personalizável, para escolas / professores 👩🏻‍🏫📄

#### Funcionalidades
* Habilitar / Desabilitar conteúdos e componentes
    * __Dados__: _Os 4 Bimestres, 5° Conceito e Resultado final_
    * __Componentes__: _Professor(a), 5° Conceito e Assinaturas_
* Mantém dados do formulário nos cookies
    * __Cabeçalho__: _Escola, Prof, Nome, N° e Ano_
* Personalização de dados para o cálculo das notas do boletim escolar
    * _Nota de aprovação, Nota de recuperação, Porcentagem minima de frequência para aprovação e Aulas dadas (por matéria)_
* Gera imagens do boletim escolar feito pelo usuário
    * __Modelo do nome do arquivo__:
        <p>{nome}__n°{numero}__{ano}__{hora}h-{minuto}m-{segundos}s.png</p>
        <p>matheus-de-oliveira-vidigal-peixoto-dias__n°27__3°-a__19h-45m-36s.png</p>
* Matérias podem ser adicionadas e removidas
* Mudança de cores do tema da aplicação e do boletim escolar

## Tecnologias 🚀
Esse projeto foi desenvolvido com as seguintes tecnologias:
- [ReactJS](https://pt-br.reactjs.org)
- [Typescript](https://www.typescriptlang.org)
- [NextJS](https://nextjs.org)
- [TailwindCSS](https://tailwindcss.com)

## Layout 🚧
### Desktop Screenshot
<div style="display: flex; flex-direction: 'column'; align-items: 'center';">
<!-- Responsive, 1366 x 768, 50% (Laptop L - 1366px) -->
    <img width="400px" src=".github/home-desktop-dark.png">
    <img width="400px" src=".github/home-desktop-light.png">
</div>

### Mobile Screenshot
<div style="display: flex; flex-direction: 'row';">
<!-- Responsive, 320 x 711, 75% (Mobile X11T - 320px) -->
    <img width="180px" src=".github/home-mobile-dark.png">
    <img width="180px" src=".github/home-mobile-light.png">
</div>

## Rodando o projeto 🚴🏻‍♂️
#### "Só vou dar uma olhadinha...":
  <a href="https://boletim-escolar.vercel.app">👩🏻‍🏫 Site hospedado na Vercel 📄</a>

#### Na sua maquina:
```bash
# Clone o repositório
$ git clone https://github.com/NyctibiusVII/BoletimEscolar.git

# Acesse a pasta do projeto no terminal
$ cd BoletimEscolar

# Instale as dependências com o gerenciador de pacotes de sua preferência
$ npm install   /   yarn add

# Execute o projeto
$ npm run dev   /   yarn dev

# O projeto roda na porta: 3000

# Acesse http://localhost:$PORT *Ex: Cuidado para não ligar dois ou mais projetos na mesma porta.
```

## Contribuição 💭
Confira a página de [contribuição](./CONTRIBUTING) para ver como começar uma discussão e começar a contribuir.

## Licença ⚖️
Este projeto está sob a licença do MIT. Veja o arquivo [LICENSE](https://github.com/NyctibiusVII/BoletimEscolar/blob/main/LICENSE) para mais detalhes.

## Contribuidores 🦸🏻‍♂️
<a href="https://github.com/NyctibiusVII/BoletimEscolar/graphs/contributors">
    <img src="https://contributors-img.web.app/image?repo=NyctibiusVII/BoletimEscolar&max=500" alt="Lista de contribuidores" width="15%"/>
</a>

<br/>
<br/>

###### Feito com ❤️ por Matheus Vidigal 👋🏻 [Entre em contato!](https://www.linkedin.com/in/matheus-vidigal-nyctibiusvii)

CHANGELOG
# Change Log

All notable changes in the "BoletimEscolar" project will be documented in this file.

#### How does versioning work?
- Version tag (v)
- [Major].[Deploy].[Patch]
- `v1.10.100`

Check [Keep a Changelog](https://keepachangelog.com/pt-BR/1.0.0/) for recommendations on how to structure this file.

## [Unreleased]
- Initial commit from Create Next App
- chore: Project setup
- feat: Bi-monthly subscription component
- chore: Unform
- feat: School and student information
- feat: Table component
- feat: Turning html into image
- refactor: Functions and styling

## [1.1.9] - 2023-05-01
### Removed
- `hello.ts`
### Changed
- `.gitignore`

## [1.2.10, 1.2.11] - 2023-05-01
### Changed
- Changing classes to add custom colors in grades component in `index.tsx` file
- `globals.css`

## [1.2.12] - 2023-05-03
### Changed
- Refactoring the css to tailwindcss in `tailwind.config.js` file
- `globals.css`

## [1.3.13] - 2023-05-03
### Added
- `components/details/index.tsx`
### Changed
- Added a sidebar in `pages/index.tsx` file
- `components/form/input/index.tsx`
- `types.ts`
- `package.json`
- `yarn.lock`

## [1.4.14, 1.4.15, 1.5.16, 1.6.17] - 2023-05-04
### Removed
- `favicon.ico`
- `next.svg`
- `vercel.svg`
### Added
- `favicon.svg`
- `HomePage-1366x768.png`
- `components/sidebar/index.tsx`
- `context/GenerateImageContext.tsx`
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- `context/SidebarContext.tsx`
- `useSchoolReport.tsx`
- `useSchoolReportConfig.tsx`
- `useSidebar.tsx`
- `.editorconfig`
- `next-sitemap.config`
### Changed
- `components/details/index.tsx`
- `components/input/index.tsx`
- `components/sidebar/index.tsx`
- `context/SchoolReportConfigContext.tsx`
- `types.ts`
- `_documents.tsx`
- `_app.tsx`
- `pages/index.tsx`
- `next-sitemap.config`
- `package.json`
- `yarn.lock`

## [1.7.18] - 2023-05-06
### Changed
- `components/details/index.tsx`
- `components/input/index.tsx`
- Adding toggle theme in `components/sidebar/index.tsx` file
- Adding typing in `types.ts` file
- Applying provider theme in `_app.tsx` file
- `globals.css`
- `package.json`
- `tailwind.config.js`
- `yarn.lock`

## [1.8.19, 1.9.20, 1.10.21] - 2023-05-07
### Changed
- Adding 'Habilitar / Desabilitar' Component in `components/sidebar/index.tsx` file
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- Applying provider in `_app.tsx` file
- Fixing the calculation of the results in `pages/index.tsx` file

## [1.10.22, 1.11.23, 1.12.24, 1.13.25] - 2023-05-10
### Removed
- `assets/cookie-texture2.jpg`
### Added
- `assets/cookie-texture.jpg`
- `assets/cookie-texture2.jpg`
- `components/modal/cookieConcent/index.tsx`
- `utils/providerTree.tsx`
### Changed
- `components/sidebar/index.tsx`
- `context/GenerateImageContext.tsx`
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- `types.ts`
- `_app.tsx`
- `pages/index.tsx`
- `package.json`
- `tailwind.config.js`
- `yarn.lock`

## [1.14.26, 1.15.27, 1.16.28] - 2023-05-11
### Added
- `LICENSE`
- `CONTRIBUTING`
- Images in `.github` folder
### Changed
- Adding 'Imagens' Component in `components/sidebar/index.tsx` file
- `context/GenerateImageContext.tsx`
- `types.ts`
- `README.md`
- `.editorconfig`

## [1.17.29] - 2023-05-12
### Added
- `.well-known/security.txt`
- `manifest.json`
- `.nvmrc`
### Changed
- `package.json`
- `yarn.lock`

## [1.17.30, 1.17.31, 1.17.32] - 2023-05-22
### Changed
- `context/GenerateImageContext.tsx`
- Added modal in `pages/index.tsx` file
- `package.json`
- Fixing color error in `tailwind.config.js` file
- `yarn.lock`

## [1.17.33, 1.17.34, 1.17.35, 1.18.36, 1.18.37, 1.19.38] - 2023-05-24
### Added
- `useSwalTheme.tsx`
- `utils/reactIconsImports.tsx`
- `utils/converterText.tsx`
### Changed
- `components/sidebar/index.tsx`
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- `types.ts`
- `pages/index.tsx`
- `globals.css`
- `README.md`

## [1.19.39] - 2023-05-26
### Added
- `components/infoIcon/index.tsx`
### Changed
- `components/sidebar/index.tsx`
- `types.ts`

## [1.19.40, 1.19.41, 1.19.42] - 2023-05-31
### Added
- `context/LoadingContext.tsx`
-  Saving data to LocalStorage in `context/LocalStorageContext.tsx` file
- `useLoading.tsx`
- `utils/isObject.tsx`
### Changed
- `context/GenerateImageContext.tsx`
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- Custom subjects can be deleted in inactiveSubjects in `context/SidebarContext.tsx` file
- System theme in `useSwalTheme.tsx` file
- `types.ts`
- `_app.tsx`
- `pages/index.tsx`

## [2.20.43] - 2023-06-01
### Changed
- `components/sidebar/index.tsx`
- `context/GenerateImageContext.tsx`
- `utils/reactIconsImports.ts`
- `types.ts`
- `package.json`
- `yarn.lock`

## [2.20.44] - 2023-06-07
### Added
- `components/skeleton/pages/SkeletonHome/index.tsx`
- `components/skeleton/index.tsx`
### Changed
- `types.ts`
- `pages/index.tsx`
- `globals.css`

## [2.20.45, 2.20.46, 2.20.47, 2.20.48, 2.20.49, 2.20.50, 2.20.51, 2.20.52, 2.20.53, 2.20.54, 2.20.55, 2.20.56, 2.20.57, 2.20.58, 2.20.59, 2.20.60, 2.20.61, 2.21.62, 2.22.63] - 2023-07-04
### Added
- `components/schoolReport/index.tsx`
- `components/skeleton/pages/SkeletonHome/index.tsx`
- `components/skeleton/index.tsx`
- `useTheme.tsx`
- `utils/generateUniqueRandomNumber.ts`
### Changed
- `components/details/index.tsx`
- `components/sidebar/index.tsx`
- `components/infoIcon/index.tsx`
- `components/input/index.tsx`
- `components/modal/cookieConcent/index.tsx`
- `components/skeleton/pages/SkeletonHome/index.tsx`
- `components/skeleton/index.tsx`
- `context/LocalStorageContext.tsx`
- `context/SchoolReportConfigContext.tsx`
- `context/SchoolReportContext.tsx`
- `utils/generateUniqueRandomNumber.ts`
- `useSwalTheme.tsx`
- `types.ts`
- `pages/index.tsx`
- `globals.css`
- `.gitignore`

## [3.23.64] - 2023-07-08
### Removed
- `utils/generateUniqueRandomNumber.ts`
### Changed

src/utils/isObject.ts
import {
    ActiveQuarter,
    FilesImage,
    MaintainReportCardData,
    Matter,
    SchoolReportColors
} from '@/interfaces/types'

export const isActiveQuarter = (obj: any): obj is ActiveQuarter => {
    return typeof obj === 'object'
        && 'firstQuarter'  in obj
        && 'secondQuarter' in obj
        && 'thirdQuarter'  in obj
        && 'fourthQuarter' in obj
}
export const isFilesImage = (obj: any): obj is FilesImage => {
    return Array.isArray(obj)
        && obj.every(item =>
                typeof item === 'object'
                && 'name' in item
                && 'imageData' in item
            )
}
export const isMaintainReportCardData = (obj: any): obj is MaintainReportCardData => {
    return typeof obj === 'object'
        && 'school'              in obj
        && 'teacher'             in obj
        && 'studentName'         in obj
        && 'studentNumber'       in obj
        && 'studentYearAndClass' in obj
}
export const isSubjects = (obj: any): obj is Matter[] => {
    return Array.isArray(obj) && obj.every(item => typeof item === 'string')
}
export const isSchoolReportColors = (obj: any): obj is SchoolReportColors => {
    return typeof obj === 'object'
        && 'card'              in obj
        && 'border'            in obj
        && 'clippingBorder'    in obj
        && 'signatures'        in obj
        && 'text'              in obj
        && 'insufficientGrade' in obj
        && 'enoughGrade'       in obj
}

next-sitemap.config.js
/** @type {import('next-sitemap').IConfig} */
module.exports = {
  siteUrl: process.env.SITE_URL || 'https://boletim-escolar-nyctibiusvii.vercel.app',
  changefreq: 'daily',
  priority: 0.7,
  sitemapSize: 5000,
  generateRobotsTxt: true,
  exclude: ['/__tests__'],
  alternateRefs: [],
  // - Default transformation function
  transform: async (config, path) => {
    return {
      loc: path, // => - This will be exported as http(s)://<config.siteUrl>/<path>
      changefreq: config.changefreq,
      priority: config.priority,
      lastmod: config.autoLastmod ? new Date().toISOString() : undefined,
      alternateRefs: config.alternateRefs ?? [],
    }
  },
  robotsTxtOptions: {
    policies: [
      {
        userAgent: '*',
        allow: ['/'/*, '/404'*/],
        disallow: ['/__tests__']
      }
    ]
  }
}


src/interfaces/types.ts
export type FilesImage = {
    name: string
    imageData: string
}[]
export type MaintainReportCardData = {
    school:                     boolean
    teacher:                    boolean
    studentName:                boolean
    studentNumber:              boolean
    studentYearAndClass:        boolean
    academicRecordGrades:       boolean
    academicRecordAbsences:     boolean
    academicRecordTotalClasses: boolean
}
export type ActiveQuarter = {
    firstQuarter:  boolean
    secondQuarter: boolean
    thirdQuarter:  boolean
    fourthQuarter: boolean
}
export type Bimester = {
    firstQuarter:  number
    secondQuarter: number
    thirdQuarter:  number
    fourthQuarter: number
}
export type SchoolReportColors = {
    card:              string
    border:            string
    clippingBorder:    string
    signatures:        string
    text:              string
    insufficientGrade: string
    enoughGrade:       string
}
export type SchoolReportStyles = { [key: string]: React.CSSProperties }
export type Matter =
  | 'Português'
  | 'Matemática'
  | 'História'
  | 'Geografia'
  | 'Física'
  | 'Química'
  | 'Biologia'
  | 'Ciências'
  | 'Filosofia'
  | 'Sociologia'
  | 'Inglês'
  | 'Educação Física'
  | 'Artes'
  | 'Ensino Religioso'
  | string
export enum SubjectSituation {
    APPROVED           = 'Aprovado',
    RECOVERY           = 'Em recuperação',
    FAILED_FOR_ABSENCE = 'Reprovado por falta',
    DISAPPROVED        = 'Reprovado'
}
export enum Concept {
    A = 'A',
    B = 'B',
    C = 'C',
    D = 'D'
}
export interface AcademicRecord {
    grades:   Bimester
    absences: Bimester
    concept:  Concept
    totalClasses:  number
    totalAbsences: number
    finalResult:   SubjectSituation
}
export interface StudentAcademicRecord {
    [subject: Matter]: AcademicRecord
}
export interface Student {
    name:         string
    number:       number
    yearAndClass: string
}
export interface SchoolReportType {
    school:       string
    teacher:      string
    academicYear: number
    student:               Student
    studentAcademicRecord: StudentAcademicRecord
}
/* Components/ColorPicker */
type SchoolReportKeys = keyof SchoolReportColors
type ColorPickerType = { item: SchoolReportKeys }
/**
 * Defines the types of native input properties.
 */
export type ColorPickerProps = JSX.IntrinsicElements['button'] & ColorPickerType
/* Components/InfoIcon */
type InfoIconType = {
    topic:       string
    description: string
}
export type InfoIconProps = JSX.IntrinsicElements['svg'] & InfoIconType
/* Components/Input */
type InputType = {
    name:            string
    label?:          string
    labelStyle?:     string
    labelPosition?:  'before' | 'after'
    container?:      boolean
    containerStyle?: string
    withForm?:       boolean
}
/**
 * Defines the types of native input properties.
 */
export type InputProps = JSX.IntrinsicElements['input'] & InputType
/* Components/Details */
type DetailsType = { summary?: string, containerClassName?: string }
/**
 * Defines the types of native details properties.
 */
export type DetailsProps = JSX.IntrinsicElements['details'] & DetailsType
/* Contexts/GenerateImageContext */
export type DownloadableDataTypes =
    | string
    | ArrayBuffer
    | number[]
    | Uint8Array
    | Blob
    | NodeJS.ReadableStream
    | Promise<
        | string
        | ArrayBuffer
        | number[]
        | Uint8Array
        | Blob
        | NodeJS.ReadableStream
    >
/* Contexts/LocalStorageContext */
export type GetLocalStorage = Record<
    string,
    | string
    | number
    | boolean
    | object
    | MaintainReportCardData
    | ActiveQuarter
    | FilesImage
    | Matter[]
    | Bimester
>
export type LocalStorage_AcademicRecordType =
    | 'academic_record_total_classes'
    | 'academic_record_total_absences'
    | 'academic_record_concept'
    | 'academic_record_final_result'
/* Theme */
type Theme = 'light' | 'dark'
/**
 * Defines the default values.
 */
export enum DefaultValues {
    INPUT_TEXT   = '',
    INPUT_NUMBER = 0,
    MINIMUM_PASSING_GRADE                 = 6,
    MINIMUM_RECOVERY_GRADE                = 4,
    MINIMUM_ATTENDANCE_PERCENTAGE_TO_PASS = 25,
    KEEP_VALUES          = '{"school":true,"teacher":true,"studentName":false,"studentNumber":false,"studentYearAndClass":false,"academicRecordGrades":false,"academicRecordAbsences":false,"academicRecordTotalClasses":true}',
    BIMESTER             = '{"firstQuarter":0,"secondQuarter":0,"thirdQuarter":0,"fourthQuarter":0}',
    ACTIVE_QUARTER       = '{"firstQuarter":true,"secondQuarter":true,"thirdQuarter":true,"fourthQuarter":true}',
    ACTIVE_SUBJECTS      = '["Português","Matemática","Artes","Ciências","História","Geografia","Educação Física"]',
    INACTIVE_SUBJECTS    = '["Inglês","Física","Química","Biologia","Filosofia","Sociologia","Ensino Religioso"]',
    MATTER               = '["Português","Matemática","Artes","Ciências","História","Geografia","Educação Física","Inglês","Física","Química","Biologia","Filosofia","Sociologia","Ensino Religioso"]',
    SCHOOL_REPORT_COLORS = '{"card":"#ffffff","border":"#030712","clippingBorder":"#dc2626","signatures":"#030712","text":"#030712","enoughGrade":"#22c55e","insufficientGrade":"#dc2626"}',
    FILES_IMAGE          = '[]',
    TOTAL_CLASSES  = 56,
    TOTAL_ABSENCES = 0,
    CONCEPT      = Concept.D,
    FINAL_RESULT = SubjectSituation.DISAPPROVED
}
/* Component/Skeleton */
type ChildrenProps = { children: React.ReactNode }
type OptionalChildrenProps = { children?: React.ReactNode }
type OptionalClassNameProps = { className?: string }
type SkeletonSimpleComponentsTypes = 'card' | 'avatar'
type SkeletonComplexComponentsTypes = 'container' | 'item' | 'list'
type SkeletonComponentsTypes = SkeletonSimpleComponentsTypes & SkeletonComplexComponentsTypes
type SkeletonSizeOption = 'fit' | 'sm' | 'md' | 'lg' | 'xl' | 'full'
type OptionalSkeletonSize = { size?: SkeletonSizeOption }
type SkeletonSize = { size: SkeletonSizeOption }
type OptionalSkeletonSizeWH = {
    width?:  SkeletonSizeOption
    height?: SkeletonSizeOption
}
type SkeletonSizeWH = {
    width:  SkeletonSizeOption
    height: SkeletonSizeOption
}
export type SizeResolverProps = SkeletonSizeWH & { ComponentType: SkeletonComplexComponentsTypes }
export type SizeMappingType = Record<SkeletonSizeOption, string>
export type ComponentMappingType = Record<SkeletonComplexComponentsTypes, string>
export type ProportionalSizeResolverProps = SkeletonSizeOption
export type ProportionalSizeMappingType = Record<SkeletonSizeOption, string>
export type SkeletonComponentsProps = OptionalSkeletonSizeWH & OptionalClassNameProps
export type SkeletonProportionalComponentsProps = OptionalSkeletonSize & OptionalClassNameProps
export type SkeletonContainerProps = SkeletonComponentsProps & ChildrenProps
export type SkeletonListProps = SkeletonComponentsProps & { count?: number }
export type SkeletonSidebarProps = { isOpen: boolean, theme: Theme }
export interface SkeletonProps {
    Container: (props: SkeletonContainerProps) => JSX.Element
    Card:      (props: SkeletonProportionalComponentsProps) => JSX.Element
    Item:      (props: SkeletonComponentsProps) => JSX.Element
    List:      (props: SkeletonListProps) => JSX.Element
    Avatar:    (props: SkeletonProportionalComponentsProps) => JSX.Element
}

tailwind.config.js
/** @type {import('tailwindcss').Config} */
module.exports = {
  content: [
    './src/pages/**/*.{js,ts,jsx,tsx,mdx}',
    './src/components/**/*.{js,ts,jsx,tsx,mdx}',
    './src/app/**/*.{js,ts,jsx,tsx,mdx}'
  ],
  darkMode: 'class',
  theme: {
    colors: {
      transparent: 'transparent',
      'shadow-5' : '#0000000d',
      'shadow-15' : '#00000026',
      'shadow-30' : '#0000004d',
      black: '#000000',
      white: '#ffffff',
      slate: {
        50: '#f8fafc',
        100: '#f1f5f9',
        200: '#e2e8f0',
        300: '#cbd5e1',
        400: '#94a3b8',
        500: '#64748b',
        600: '#475569',
        700: '#334155',
        800: '#1e293b',
        900: '#0f172a',
        950: '#020617',
      },
      gray: {
        50: '#f9fafb',
        100: '#f3f4f6',
        200: '#e5e7eb',
        300: '#d1d5db',
        400: '#9ca3af',
        500: '#6b7280',
        600: '#4b5563',
        700: '#374151',
        800: '#1f2937',
        900: '#111827',
        950: '#030712',
      },
      zinc: {
        50: '#fafafa',
        100: '#f4f4f5',
        200: '#e4e4e7',
        300: '#d4d4d8',
        400: '#a1a1aa',
        500: '#71717a',
        600: '#52525b',
        700: '#3f3f46',
        800: '#27272a',
        900: '#18181b',
        950: '#09090b',
      },
      neutral: {
        50: '#fafafa',
        100: '#f5f5f5',
        200: '#e5e5e5',
        300: '#d4d4d4',
        400: '#a3a3a3',
        500: '#737373',
        600: '#525252',
        700: '#404040',
        800: '#262626',
        900: '#171717',
        950: '#0a0a0a',
      },
      stone: {
        50: '#fafaf9',
        100: '#f5f5f4',
        200: '#e7e5e4',
        300: '#d6d3d1',
        400: '#a8a29e',
        500: '#78716c',
        600: '#57534e',
        700: '#44403c',
        800: '#292524',
        900: '#1c1917',
        950: '#0c0a09',
      },
      red: {
        50: '#fef2f2',
        100: '#fee2e2',
        200: '#fecaca',
        300: '#fca5a5',
        400: '#f87171',
        500: '#ef4444',
        600: '#dc2626',
        700: '#b91c1c',
        800: '#991b1b',
        900: '#7f1d1d',
        950: '#450a0a',
      },
      orange: {
        50: '#fff7ed',
        100: '#ffedd5',
        200: '#fed7aa',
        300: '#fdba74',
        400: '#fb923c',
        500: '#f97316',
        600: '#ea580c',
        700: '#c2410c',
        800: '#9a3412',
        900: '#7c2d12',
        950: '#431407',
      },
      amber: {
        50: '#fffbeb',
        100: '#fef3c7',
        200: '#fde68a',
        300: '#fcd34d',
        400: '#fbbf24',
        500: '#f59e0b',
        600: '#d97706',
        700: '#b45309',
        800: '#92400e',
        900: '#78350f',
        950: '#451a03',
      },
      yellow: {
        50: '#fefce8',
        100: '#fef9c3',
        200: '#fef08a',
        300: '#fde047',
        400: '#facc15',
        500: '#eab308',
        600: '#ca8a04',
        700: '#a16207',
        800: '#854d0e',
        900: '#713f12',
        950: '#422006',
      },
      lime: {
        50: '#f7fee7',
        100: '#ecfccb',
        200: '#d9f99d',
        300: '#bef264',
        400: '#a3e635',
        500: '#84cc16',
        600: '#65a30d',
        700: '#4d7c0f',
        800: '#3f6212',
        900: '#365314',
        950: '#1a2e05',
      },
      green: {
        50: '#f0fdf4',
        100: '#dcfce7',
        200: '#bbf7d0',
        300: '#86efac',
        400: '#4ade80',
        500: '#22c55e',
        600: '#16a34a',
        700: '#15803d',
        800: '#166534',
        900: '#14532d',
        950: '#052e16',
      },
      emerald: {
        50: '#ecfdf5',
        100: '#d1fae5',
        200: '#a7f3d0',
        300: '#6ee7b7',
        400: '#34d399',
        500: '#10b981',
        600: '#059669',
        700: '#047857',
        800: '#065f46',
        900: '#064e3b',
        950: '#022c22',
      },
      teal: {
        50: '#f0fdfa',
        100: '#ccfbf1',
        200: '#99f6e4',
        300: '#5eead4',
        400: '#2dd4bf',
        500: '#14b8a6',
        600: '#0d9488',
        700: '#0f766e',
        800: '#115e59',
        900: '#134e4a',
        950: '#042f2e',
      },
      cyan: {
        50: '#ecfeff',
        100: '#cffafe',
        200: '#a5f3fc',
        300: '#67e8f9',
        400: '#22d3ee',
        500: '#06b6d4',
        600: '#0891b2',
        700: '#0e7490',
        800: '#155e75',
        900: '#164e63',
        950: '#083344',
      },
      sky: {
        50: '#f0f9ff',
        100: '#e0f2fe',
        200: '#bae6fd',
        300: '#7dd3fc',
        400: '#38bdf8',
        500: '#0ea5e9',
        600: '#0284c7',
        700: '#0369a1',
        800: '#075985',
        900: '#0c4a6e',
        950: '#082f49',
      },
      blue: {
        50: '#eff6ff',
        100: '#dbeafe',
        200: '#bfdbfe',
        300: '#93c5fd',
        400: '#60a5fa',
        500: '#3b82f6',
        600: '#2563eb',
        700: '#1d4ed8',
        800: '#1e40af',
        900: '#1e3a8a',
        950: '#172554',
      },
      indigo: {
        50: '#eef2ff',
        100: '#e0e7ff',
        200: '#c7d2fe',
        300: '#a5b4fc',
        400: '#818cf8',
        500: '#6366f1',
        600: '#4f46e5',
        700: '#4338ca',
        800: '#3730a3',
        900: '#312e81',
        950: '#1e1b4b',
      },
      violet: {
        50: '#f5f3ff',
        100: '#ede9fe',
        200: '#ddd6fe',
        300: '#c4b5fd',
        400: '#a78bfa',
        500: '#8b5cf6',
        600: '#7c3aed',
        700: '#6d28d9',
        800: '#5b21b6',
        900: '#4c1d95',
        950: '#2e1065',
      },
      purple: {
        50: '#faf5ff',
        100: '#f3e8ff',
        200: '#e9d5ff',
        300: '#d8b4fe',
        400: '#c084fc',
        500: '#a855f7',
        600: '#9333ea',
        700: '#7e22ce',
        800: '#6b21a8',
        900: '#581c87',
        950: '#3b0764',
      },
      fuchsia: {
        50: '#fdf4ff',
        100: '#fae8ff',
        200: '#f5d0fe',
        300: '#f0abfc',
        400: '#e879f9',
        500: '#d946ef',
        600: '#c026d3',
        700: '#a21caf',
        800: '#86198f',
        900: '#701a75',
        950: '#4a044e',
      },
      pink: {
        50: '#fdf2f8',
        100: '#fce7f3',
        200: '#fbcfe8',
        300: '#f9a8d4',
        400: '#f472b6',
        500: '#ec4899',
        600: '#db2777',
        700: '#be185d',
        800: '#9d174d',
        900: '#831843',
        950: '#500724',
      },
      rose: {
        50: '#fff1f2',
        100: '#ffe4e6',
        200: '#fecdd3',
        300: '#fda4af',
        400: '#fb7185',
        500: '#f43f5e',
        600: '#e11d48',
        700: '#be123c',
        800: '#9f1239',
        900: '#881337',
        950: '#4c0519',
      }
    },
    opacity: {
      '0': '0',
      '20': '0.2',
      '40': '0.4',
      '60': '0.6',
      '80': '0.8',
      '100': '1'
    },
    extend: {
      backgroundImage: {
        'cookie-texture': `url('/assets/cookie-texture.jpg')`
      }
    }
  }
}
